formbuilder/components/builder/EditableField.js

```diff
@@ -39,16 +48,23 @@ class FieldPropertiesEditor extends Component {
       ...this.state.editedSchema,
       name: this.state.name
     };
+
     return (
       <div className="panel panel-default field-editor">
-        <div className="panel-heading">
-          <strong>Edit {name}</strong>
-          <button type="button" className="close-btn" onClick={onCancel} aria-label="Close">
-            close <i className="glyphicon glyphicon-remove-sign"/>
-          </button>
-          <button type="button" className="close-btn" onClick={onDelete} aria-label="Delete">
-            delete <i className="glyphicon glyphicon-trash"/>
-          </button>
+        <div className="panel-heading clearfix">
+            <strong className="panel-title">Edit {name}</strong>
+
+            <ButtonToolbar className="pull-right">
+              <FieldListDropdown bsStyle="link" {...this.props} currentFieldName={this.props.name}>
```

Comment:
Instead of doing these tests to see if the state changed, I believe a simpler way to do would be to detect when the current field changes and trigger a different action, that would save us from a lot of if checks and special cases here and there.